crates/sui-framework/packages/sui-framework/tests/k_of_n_upgrade_policy_test.move

```diff
@@ -0,0 +1,427 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+#[test_only]
+module sui::k_of_n_upgrade_policy_test {
+    use sui::address::from_u256;
+    use sui::k_of_n_upgrade_policy::{Self, KofNUpgradeCap, ProposedUpgrade, VotingCap};
+    use sui::object::id_from_address as id;
+    use sui::package;
+    use sui::vec_set::{Self, VecSet};
+    use sui::test_scenario::{Self as test, Scenario, ctx};
+
+    #[test]
+    #[expected_failure(abort_code = sui::k_of_n_upgrade_policy::ERequiredVotesError)]
+    fun k_of_n_too_many_required_votes() {
+        let test = test::begin(@0x1);
+        let k_of_n_cap = get_k_of_n_cap(30, 5, &mut test);
+        k_of_n_upgrade_policy::make_immutable(k_of_n_cap);
+        test::end(test);
```

Comment:
For tests with expected errors, I tend to put an explicit `abort` after the statement that should error, so that I don't have to write up all the clean-up and it's clear which line is supposed to fail.

Reply:
oh thank you and good point!